### PR TITLE
Replace CanPowerDown with ToggleConditionOnOrder.

### LINKED ISSUE #14175: CanPowerDown CancelWhenDisabled should be replaced with PausableConditionalTrait.
Pausing the trait should keep the power disabled as if `CancelWhenDisabled: false`. Disabling the trait should cancel the power down.

This should wait until after #13998 has been merged.

### PR DESCRIPTION
**Pull request by pchote**
Closes #14175.

Testcase:
* Deploying the radar dome or tesla coil should prevent the powerdown cursor from working.
* Deploying the radar dome or tesla coil while powered down should remove the power down.
* Deploying the radar dome while powered down resets the powerdown toggle (remains powered up when redeployed).
* Deploying the tesla coil while powered down does not reset the powerdown toggle (powers down again when redeployed).

### CODE REVIEW

File: OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs [hunk 1]
```diff
@@ -1569,6 +1572,41 @@ internal static void UpgradeActorRules(ModData modData, int engineVersion, ref L
...
+					{
+						RenameNodeKey(cpd, "ToggleConditionOnOrder");
+
+						RenameNodeKey(cpd.Value.Nodes.FirstOrDefault(n => n.Key == "PowerupSound"), "DisabledSound");
+						RenameNodeKey(cpd.Value.Nodes.FirstOrDefault(n => n.Key == "PowerupSpeech"), "DisabledSpeech");
+						RenameNodeKey(cpd.Value.Nodes.FirstOrDefault(n => n.Key == "PowerdownSound"), "EnabledSound");
+						RenameNodeKey(cpd.Value.Nodes.FirstOrDefault(n => n.Key == "PowerdownSpeech"), "EnabledSpeech");
+						cpd.Value.Nodes.Add(new MiniYamlNode("OrderName", "PowerDown"));
+
+
```
Comment (reaperrr) [inline OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs]: > OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs:L1589: [SA1507] The code must not contain multiple blank lines in a row.

**Comment by pchote** [comment]
Fixed.

File: mods/ra/rules/defaults.yaml [hunk 2]
```diff
@@ -1097,16 +1097,20 @@
...
+		OrderName: PowerDown
 	WithDecoration@POWERDOWN:
 		Image: poweroff
 		Sequence: offline
 		Palette: chrome
 		RequiresCondition: powerdown
 		ReferencePoint: Center
+	PowerMultiplier@POWERDOWN:
+		RequiresCondition: powerdown
+		Modifier: 0
```
Comment (reaperrr) [inline mods/ra/rules/defaults.yaml]: This makes several yaml entries rendundant (in all our mods with the power-down option), for example on `IRON` in RA:
```
	Power:
		RequiresCondition: !powerdown
```

**Review by reaperrr — DISMISSED** [review]
Code looks good and all testcases work

File: mods/ra/rules/defaults.yaml (hunk 2 above)
Comment (pchote) [reply mods/ra/rules/defaults.yaml]: Done.

**Review by reaperrr — DISMISSED** [review]

File: OpenRA.Mods.Common/Traits/Conditions/ToggleConditionOnOrder.cs [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+			if (!IsTraitDisabled && !IsTraitPaused && order.OrderString == Info.OrderName)
+			{
+				enabled = !enabled;
+				SetCondition(self, enabled);
+			}
+		}
+
+		protected override void TraitDisabled(Actor self)
+		{
+			// Pausing the trait cancels the condition
```
Comment (obrakmann) [inline OpenRA.Mods.Common/Traits/Conditions/ToggleConditionOnOrder.cs]: Pausing -> Disabling?

**Review by obrakmann — DISMISSED** [review]
Looks good to me, 👍 

Please remove the testcase

**Comment by pchote** [comment]
Done

**Review by obrakmann — APPROVED** [review]
👍 stands, no change to the code

**Comment by obrakmann** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/992b48af1e1f56bcfd6d5e84f3c7b2d99adc1922)

**Pull request merged**